dan/webserver/webserver.py

```diff
@@ -68,42 +66,34 @@ async def chat(request, ws):
                 }
                 
                 # store message
-                messages.append(message)
-                if len(messages) > 20:  # keep only last 20 messages
-                    messages.pop(0)
+                __message_append(message)
+                
                 await _send_to_all_clients(message)
             
             elif data['type'] == 'pin_message':
-                #create pinned message object
                 pinned_message = {
```

Comment:
added check for to see if message was already pinned, as well as making sure original message author is retained when pinning.